src/main/java/chess/domain/Board.java

```diff
@@ -59,45 +57,67 @@ private void validateSameTeam(Piece sourcePiece, Piece targetPiece) {
     }
 
     private void validateMovement(Position source, Position target) {
-        Piece sourcePiece = squares.get(source);
-        if (!sourcePiece.canMove(source, target)) {
+        if (!canMove(source, target)) {
             throw new IllegalArgumentException(MOVE_FAIL_EXCEPTION_MESSAGE);
         }
     }
 
-    private void validatePieceRole(Position source, Position target) {
+    private boolean canMove(Position source, Position target) {
         Piece sourcePiece = squares.get(source);
-        if (!sourcePiece.isRoleOf(Role.KNIGHT)) {
-            validateCollision(source, target);
+        if (sourcePiece.isRoleOf(Role.KNIGHT)) {
+            return sourcePiece.canMove(source, target);
         }
         if (sourcePiece.isRoleOf(Role.PAWN)) {
-            validatePawnMove(source, target);
+            return sourcePiece.canMove(source, target) && pawnCanMove(source, target);
         }
+        return sourcePiece.canMove(source, target) && isEmptyRoute(source, target);
     }
 
-    private void validateCollision(Position source, Position target) {
+    private boolean pawnCanMove(Position source, Position target) {
```

Comment:
```suggestion
    private boolean canPawnMove(Position source, Position target) {
```

사소하지만, 영문법상 요게 더 맞을 것 같아요 👀

Reply:
영어가 너무 약하네요 😂